maven-extension/src/main/java/io/opentelemetry/maven/OtelLifecycleParticipant.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.maven;
+
+import org.apache.maven.AbstractMavenLifecycleParticipant;
+import org.apache.maven.MavenExecutionException;
+import org.apache.maven.execution.ExecutionListener;
+import org.apache.maven.execution.MavenSession;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Add the {@link OtelExecutionListener} to the lifecycle of the MAven execution */
+@Component(role = AbstractMavenLifecycleParticipant.class)
+public class OtelLifecycleParticipant extends AbstractMavenLifecycleParticipant {
```

Comment:
If I'm not mistaken, this is the entry-point so hopefully this is the only class we need to make `public`